### PR TITLE
[14.0][FIX] hr_expense_tier_validation: disallow editable expense from the expense report is under validation

### PR DESCRIPTION
**Pull request by ps-tubtim**

### CODE REVIEW

File: hr_expense_tier_validation/models/hr_expense.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        for val in vals:
+            if val not in exceptions:
+                return False
+        return True
+
+    def write(self, vals):
+        for rec in self:
+            if (
+                rec.sheet_id.state
+                not in ["submit", "approve", "post", "done", "cancel"]
```
Comment (kittiu) [inline hr_expense_tier_validation/models/hr_expense.py]: Do we need this 2 lines?

Comment (ps-tubtim) [reply hr_expense_tier_validation/models/hr_expense.py]: I will change it to `rec.sheet_id.state == "draft"`.

**Comment by kittiu** [comment]
Functional tested and code reviewed. LGTM.

But please add test to cover case,
* When in validation process, do not allow to edit also the hr.expense

**Review by kittiu — APPROVED** [review]

**Review by Saran440 — APPROVED** [review]
Functional Test :+1:

**Comment by Saran440** [comment]
add test script for cover all case

**Review by nimarosa — APPROVED** [review]

**Pull request merged**